features/setting_constraints/matching_arguments.feature

```diff
@@ -72,6 +72,36 @@ Feature: Matching arguments
       | expected: (a collection containing exactly 1 and 2)           |
       | got: ([1, 3])                                                 |
 
+  Scenario: Using satisfy for complex custom expecations
+    Given a file named "rspec_satisfy_spec.rb" with:
+      """ruby
+      RSpec.describe "Using satisfy for complex custom expecations" do
+        let(:dbl) { double }
+
+        def a_b_c_equals_5
+          satisfy { |data| data[:a][:b][:c] == 5 }
+        end
+
+        it "passes when the expectation is true" do
+          expect(dbl).to receive(:foo).with(a_b_c_equals_5)
+          dbl.foo({ :a => { :b=> { :c => 5 } } })
+        end
+
+        it "fails when the expectation is false" do
+          expect(dbl).to receive(:foo).with(a_b_c_equals_5)
+          dbl.foo({ :a => { :b=> { :c => 3 } } })
+        end
+      end
+      """
+    When I run `rspec rspec_satisfy_spec.rb`
+    Then it should fail with the following output:
+      | 2 examples, 1 failure                                         |
+      |                                                               |
+      | Failure/Error: dbl.foo({ :a => { :b=> { :c => 3 } } })        |
+      | #<Double (anonymous)> received :foo with unexpected arguments |
+      | expected: (satisfy expression `data[:a][:b][:c] == 5`)        |
```

Comment:
Wondering if some option would make cucumber's error message more verbose on CI to avoid running 1.8.7 locally.
```
 expected ".F\n\nFailures:\n\n  1) Using satisfy for complex custom expecations fails when the expectation is f...y_spec.rb:13 # Using satisfy for complex custom expecations fails when the expectation is false\n\n" to include "expected: (satisfy expression `data[:a][:b][:c] == 5`)" (RSpec::Expectations::ExpectationNotMetError)
./features/step_definitions/additional_cli_steps.rb:20:in `/^it should fail with the following output:$/'
features/setting_constraints/matching_arguments.feature:97:in `Then it should fail with the following output:'
```

If not, [this](https://github.com/rbenv/homebrew-tap) helped me last time.